feature-pack/src/main/resources/modules/system/layers/base/org/jboss/weld/core/main/module.xml

```diff
@@ -51,5 +51,10 @@
         <module name="org.jboss.weld.spi" />
         <module name="org.jboss.logging" />
         <module name="sun.jdk" optional="true" />
+        <system export="false">
+            <paths>
+                <path name="com/sun/org/apache/bcel/internal/classfile"/>
```

Comment:
no there is no reason to restrict them, if we need them they should be there.
All classes that are part of specific jdk should be part of that module, that way you don't pollute your own modules with jdk specific package includes